src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -266,6 +268,13 @@
 
     }
 
+    private static boolean currentUserHasRunScriptsPermission() {
+        return Jenkins.getActiveInstance()
+                .getAuthorizationStrategy()
+                .getACL(User.current())
```

Comment:
this to me looks incorrect; it returns the ACL for the `User` object - which the current user will likely always have full control of given that it is the current users `User`.

Should check `Jenkins.getActiveInstance().getACL().hasPermission(Jenkins.RUN_SCRIPTS)`